components/AccountWidget/AccountWidget.jsx

```diff
@@ -48,4 +49,20 @@ const mapDispatchToProps = (dispatch) => ({
   getPlayer: (playerId) => dispatch(getPlayer(playerId)),
 });
 
-export default connect(mapStateToProps, mapDispatchToProps)(AccountWidget);
+const getData = props => {
+  props.getPlayer(props.user.account_id);
+};
+
+class RequestLayer extends React.Component {
+  componentWillUpdate(nextProps) {
+    if (nextProps.user && nextProps.user.account_id) {
+      getData(nextProps);
+    }
```

Comment:
this is going to fire every time the component updates, even if the props are the same. ComponentWillUpdate doesn't mean that every prop changed, just that something changed. You should check that nextProps.user.account_id !== this.props.user.account_id, otherwise you will fire this async request every rerender (I think). Also, I think it makes more sense to just call this.props.getPlayer(nextProps.user.account_id) since your factored out function is just a single line. Most importantly though, this isn't getting the user. This is getting player data. Player data is used by the /players pages. Metadata stores the user info. You need to call getMetadata here, not getPlayer. Look at the main index.jsx page. It dispatches getMetadata on app load. So I don't know if you ever even need to dispatch any actions since going to log in will set the cookie and then when the user comes back, it will cause a page refresh and getMetadata. So I would remove this entire RequestLayer component as well as mapDispatchToProps.